extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -47,4 +101,65 @@ public fun MapboxMap(
     },
     modifier = modifier,
   )
+
+  val parentComposition = rememberCompositionContext()
+  val currentContent by rememberUpdatedState(content)
+
+  LaunchedEffect(Unit) {
+    val composition = Composition(
+      MapApplier(mapView), parentComposition
+    ).apply {
+      setContent {
+        ComposeNode<MapboxMapNode, MapApplier>(
+          factory = {
+            MapboxMapNode(
+              mapView,
+              onMapClickListener,
+              onMapLongClickListener
+            )
+          },
+          update = {
+            // input arguments updater
+            update(onMapClickListener) { listener ->
```

Comment:
nit:
maybe move to separate small functions? will be easy to read and understand code

Reply:
refactored